src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.Analysis.Tree.cs

```diff
@@ -260,6 +290,11 @@ public override BoundNode VisitBlock(BoundBlock node)
 
                 public override BoundNode VisitCatchBlock(BoundCatchBlock node)
                 {
+                    if (node.Locals.IsDefaultOrEmpty)
+                    {
+                        return base.VisitCatchBlock(node);
+                    }
```

Comment:
To avoid this special case in each of these methods, consider having `CreateOrReuseScope` simply return the same scope if there are no locals.